File: 프로그래머스/unrated/161989. 덧칠하기/덧칠하기.cpp
#include <string>
#include <vector>
#include <memory.h>
using namespace std;

int solution(int n, int m, vector<int> section) {
    int* wall = new int[n + 1];
    memset(wall, 0, sizeof(int) * n + 1);

    for (int i = 0; i < section.size(); i++)
    wall[section[i]] = 1;

    int cnt = 0;
    for (int i = 1; i <= n; i++) {
        if (wall[i]) {
            for (int j = i; j < i + m; j++) {
                if (j > n) break;
                wall[j] = 0;
            }
            cnt++;
        }
    }
    return cnt;
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        
        for (int i = 1, idx = 0; i <= n && idx <= section.length - 1;) {
            if (i == section[idx]) {
                i += m;
                idx++;
                answer++;
            } else if (i > section[idx]) {
                idx++;   
            } else {
                i++;
            }
        }
        
        return answer;
    }
}